sdk/nodejs/x/automation/localWorkspace.ts

```diff
@@ -61,6 +63,16 @@ export class LocalWorkspace implements Workspace {
      * Environment values scoped to the current workspace. These will be supplied to every Pulumi command.
      */
     envVars: { [key: string]: string };
+    private _pulumiVersion?: semver.SemVer;
+    /**
+     * The version of the underlying Pulumi CLI/Engine.
+     */
+    public get pulumiVersion(): semver.SemVer {
+        if (!this._pulumiVersion) {
+            throw new Error("Failed to get pulumi version.");
+        }
```

Comment:
Instead of throwing an error, would it be possible to run the `getPulumiVersion` call and populate the variable?

Reply:
The `getPulumiVersion` call happens during the construction of `LocalWorkspace`, it just doesn't complete in the constructor. Because it doesn't complete in the constructor (because it's async) I have to make `_pulumiVersion` have type `SemVer | undefined`. 

So this shouldn't ever actually be undefined unless something went wrong during the parsing, but that error should already have been thrown. This just needs to be `return this._pulumiVersion!;`. I couldn't get it to work earlier coz I was putting the `!` in the wrong spot.